src/Client_main.ml

```diff
@@ -94,14 +94,21 @@ let rec main_loop connection_info kernel =
     Log.log "Dying.\n";
     Lwt.fail Exit
 
-let main ?(args=[]) ~usage kernel =
+let main ?(args=[]) ?post_init ~usage kernel =
   let args = mk_args ~args () in
   Arg.parse
     args
     (fun s -> failwith ("invalid anonymous argument: " ^ s))
     usage;
   let connection_info = mk_connection_info () in
   let%lwt() = Lwt_io.printf "Starting kernel for `%s`\n" kernel.C.Kernel.language in
+
+    match post_init with
```

Comment:
you must wrap this in `begin match … end;` otherwise the whole `main` function will only run if `post_init` is provided :sweat_smile: